xarray/core/computation.py

```diff
@@ -396,8 +399,31 @@ def _unpack_dict_tuples(
     return out
 
 
+def _check_core_dims(signature, variable_args, name):
+    for core_dims, variable_arg in zip(signature.input_core_dims, variable_args):
+        # Check whether all the dims are on the variable. Note that we need the
+        # `hasattr` to check for a dims property, to protect against the case where
+        # a numpy array is passed in.
+        if hasattr(variable_arg, "dims") and set(core_dims) - set(variable_arg.dims):
+            # Slightly awkward design, of returning the error message. But we want to
+            # give a detailed error message, which requires inspecting the variable in
+            # the inner loop.
+            return (
+                f"Missing core dimension(s) {set(core_dims) - set(variable_arg.dims)} on `{name}`. "
+                "Either add the core dimension, or set `missing_core_dim` to `copy` or `drop`. "
```

Comment:
```suggestion
                "Either add the core dimension using `broadcast` or `expand_dims` for example, or set `missing_core_dim` to `copy` or `drop`. "
```

Reply:
Can we raise one error listing all variables with their respective missing core dimensions?

Reply:
Cool, latest version has this as the message (repeating for more vars):

```
E                   ValueError: Missing core dims from x_z on variable(s):
E
E                   Missing {'y'} on
E
E                   <xarray.Variable (x: 2, z: 3)>
E                   array([[0, 1, 2],
E                          [3, 4, 5]])
E
E                   Either add the core dimension, or if passing a dataset alternatively pass `on_missing_core_dim` as `copy` or `drop`.
```